EIPS/eip-4401.md

```diff
@@ -0,0 +1,115 @@
+    event ConsumerChanged(address indexed owner, address indexed consumer, uint256 indexed tokenId);
+
+    /// @notice Get the consumer address of an NFT
+    /// @dev The zero address indicates that there is no consumer
+    /// Throws if `_tokenId` is not a valid NFT
+    /// @param _tokenId The NFT to get the consumer address for
+    /// @return The consumer address for this NFT, or the zero address if there is none
+    function consumerOf(uint256 _tokenId) view external returns (address);
+
+    /// @notice Change or reaffirm the consumer address for an NFT
+    /// @dev The zero address indicates there is no consumer address
+    /// Throws unless `msg.sender` is the current NFT owner, an authorised
+    /// operator of the current owner or approved address
+    /// Throws if `_tokenId` is not valid NFT
+    /// @param _consumer The new consumer of the NFT
+    function changeConsumer(address _consumer, uint256 _tokenId) external;
+}
+```
+
+Every contract implementing the `EIP721Consumable` extension is free to define the permissions of a `consumer` (e.g. what are consumers allowed to do within their system) with only one exception - consumers MUST NOT be considered owners, authorised operators or approved addresses as per the EIP-721 specification. Thus, they MUST NOT be able to execute transfers & approvals.
+
+The `consumerOf(uint256 _tokenId)` function MAY be implemented as `pure` or `view`.
+
+The `changeConsumer(address _consumer, uint256 _tokenId)` function MAY be implemented as `public` or `external`.
+
+The `ConsumEIPhanged` event MUST be emitted when a consumer is changed.
+
+On every `transfer`, the consumer MUST be changed to a default address. It is RECOMMENDED for implementors to use `address(0)` as that default address.
+
+The `supportsInterface` method MUST return `true` when called with `0x953c8dfa`.
+
+## Rationale
+
+Key factors influencing the standard:
+
+- Keeping the number of functions in the interfaces to a minimum to prevent contract bloat
+- Simplicity
+- Gas Efficiency
+- Not reusing or overloading other already existing roles (e.g. owners, operators, approved addresses)
+
+### Name
+
+The chosen name resonates with the purpose of its existence. Consumers can be considered entities that utilise the token instances, without necessarily having ownership rights to it.
+
+The other name for the role that was considered was `operator`, however it is already defined and used within the `EIP-721` standard.
+
+### Restriction on the Permissions
+
+There are numerous use-cases where a distinct role for NFTs is required that MUST NOT have owner permissions. A contract that implements the consumer role and grants ownership permissions to the consumer renders this standard pointless.
+
+## Backwards Compatibility
+
+This standard is compatible with current EIP-721 standards. There are no other standards that define a similar role for NFTs and the name (`consumer`) is not used by other EIP-721 related standards.
```

Comment:
Probably worth mentioning that this isn't backward compatible with EIP-4400, due to the event name change.